framework/DataObjects/XPointSet.py

```diff
@@ -168,3 +168,41 @@ def _selectiveRealization(self,rlz):
     for var in toRemove:
       del rlz[var]
     return rlz
+
+  def _toCSV(self,fileName,start=0,**kwargs):
+    """
+      Writes this data object to CSV file (except the general metadata, see _toCSVXML)
+      @ In, fileName, str, path/name to write file
+      @ In, start, int, optional, first realization to start printing from (if > 0, implies append mode)
+      @ In, kwargs, dict, optional, keywords for options
+            Possibly includes:
+                'clusterLabel': name of variable to cluster printing by.  If included then triggers history-like printing.
+      @ Out, None
+    """
+    if not self.hierarchical and 'RAVEN_isEnding' in self.getVars():
+      keep = self._getRequestedElements(kwargs)
+      toDrop = list(var for var in self.getVars() if var not in keep)
+      # set up data to write
+      if start > 0:
+        mode = 'a'
+      else:
+        mode = 'w'
+
+      #FIXME: THIS IS EXTREMELY SLOW
+      full = self._constructHierPaths()[start-1:]
+      # set up data to write
+      mode = 'a' if start > 0 else 'w'
```

Comment:
I think we can remove this line, the mode is already defined in line 186-189

Reply:
removed